src/Stripes.test.js

```diff
@@ -0,0 +1,123 @@
+import Stripes from './Stripes';
+
+describe('Stripes', () => {
+  describe('clone', () => {
+    it('returns a matching copy', () => {
+      const s = new Stripes();
+      const c = s.clone();
+      expect(s).toEqual(c);
+    });
+
+    it('returns a matching copy plus additional properties', () => {
+      const s = new Stripes();
+      const c = s.clone({ monkey: 'bagel' });
+
+      expect(c).toMatchObject(s);
+      expect(c).toHaveProperty('monkey', 'bagel');
+    });
+  });
+
+  describe('hasInterface', () => {
+    describe('returns undefined', () => {
+      it('before discovery', () => {
+        const logger = { log: jest.fn() };
+        const s = new Stripes({ logger });
+        expect(s.hasInterface('monkey')).toBeUndefined()
+        expect(logger.log).toHaveBeenCalled();
+      });
+
+      it('before discovery.interfaces', () => {
+        const logger = { log: jest.fn() };
+        const s = new Stripes({ logger, discovery: {} });
+        expect(s.hasInterface('monkey')).toBeUndefined();
+        expect(logger.log).toHaveBeenCalled();
+      });
+
+      it('if the interface is absent', () => {
+        const logger = { log: jest.fn() };
+        const s = new Stripes({ logger, discovery: { interfaces: { 'funky': '3.0' }} });
+        expect(s.hasInterface('monkey')).toBeUndefined();
+        expect(logger.log).toHaveBeenCalledWith('interface', 'interface \'monkey\' is missing');
+      });
+    });
+
+    describe('returns true', () => {
+      it('when the interface is present and no version is provided', () => {
```

Comment:
The interface-version in this test is _provided_ (in the config) but not _required_ (in the `hasInterface` call), so maybe rename the test?